docs/contributor/migration-guide.md

```diff
@@ -8,78 +8,165 @@ It covers all necessary changes and deprecations to ensure a smooth transition.
+```bash
+modulectl -h                # general help
+modulectl create -h         # help for 'create'
+modulectl scaffold -h       # help for 'scaffold'
+modulectl list -h           # help for 'list'
+```
+
+### 1.2 Command & Flag Differences
+
+#### 1.2.1 Command Mapping
+
+| Operation                  | Kyma CLI                 | `modulectl`                              |
+| -------------------------- | ------------------------ |------------------------------------------|
+| Scaffold module template   | `kyma module init ...`   | `modulectl scaffold ...`                 |
+| Create ModuleTemplate CR   | `kyma module create ...` | `modulectl create -c <config-file>`      |
+| List templates or versions | `kyma module list ...`   | \`modulectl list \[templates releases]\` |
+| Command-specific help      | `kyma module <cmd> -h`   | `modulectl <cmd> -h`                     |
+
+#### 1.2.2 Flag & Behavior Differences
+
+| Feature                       | Deprecated in Kyma CLI                            | New in `modulectl`                                             |
+| ----------------------------- | ------------------------------------------------- | -------------------------------------------------------------- |
+| Config-file flag              | `--module-config-file`                            | `--config-file`, shortcut `-c` (all commands)                  |
+| Archive overwrite             | `--module-archive-version-overwrite`              | **Removed** (module versions are immutable)                    |
+| ModuleTemplate naming pattern | `.metadata.name = <module-name>-<channel>`        | `.metadata.name = <module-name>-<version>`                     |
+| Version label                 | `operator.kyma-project.io/module-version` label   | Populated in `.spec.version`                                   |
+| Documentation annotation      | `operator.kyma-project.io/doc-url` annotation     | Defined in `.spec.info.documentation` via `documentation:` key |
+| Manifest & DefaultCR source   | Local file references only                        | Fetched directly from GitHub release URLs                      |
+| Release channel in config     | Required `.channel` field in `module-config.yaml` | **Removed**; channel mapping managed by ModuleReleaseMeta CR   |
+
+### 1.3 Submission Process & Migration Period
+
+#### Submission Process
+
+1. **Publish Artifacts**: Release module version assets (manifest, defaultCR) on GitHub.
+2. **Submit Version Config**: Add/update `/modules/<module>/<version>/module-config.yaml`.
+3. **Update Channel Mapping**: Edit `/modules/<module>/module-releases.yaml` to map channels to versions.
+4. **Trigger Pipeline**: On PR merge, the submission pipeline:
+
+   * Validates schema, FQDN, version uniqueness
+   * Builds via `modulectl`, pushes OCI image
+   * Generates ModuleTemplate and ModuleReleaseMeta in `/kyma/kyma-modules`
+5. **ArgoCD Sync**: Deploys ModuleTemplates and ModuleReleaseMeta to KCP landscapes.
```

Comment:
Who is responsible for that? Should we write "Deploy ModuleTemplates..." as this is done by the module team or is it also done on the PR merge automatically?

Reply:
After the module is promoted , ArgoCD picks changes made by submission pipeline and deploys the ModuleTemplates and ModuleReleaseMetas relevant for the respective landscape.

Reply:
```suggestion
5. **ArgoCD Sync**: ArgoCD deploys ModuleTemplates and ModuleReleaseMeta to KCP landscapes.
```

Reply:
Maybe we can add the subject explicitly then?